systray_darwin.m

```diff
@@ -227,7 +227,23 @@ - (void) reset_menu
 
 - (void) quit
 {
-  [NSApp terminate:self];
+  // This tells the app event loop to stop after processing remaining messages.
+  [NSApp stop:self];
+  // The event loop won't return until it processes another event. https://stackoverflow.com/a/48064752/149482
+  NSPoint eventLocation = NSMakePoint(0, 0);
+  NSEvent *customEvent = [NSEvent otherEventWithType:NSEventTypeApplicationDefined
+                                            location:eventLocation
+                                       modifierFlags:0
+                                           timestamp:0
+                                        windowNumber:0
+                                             context:nil
+                                             subtype:0
+                                               data1:0
+                                               data2:0];
+
+  [NSApp postEvent:customEvent atStart:NO];
+  // We used to terminate here instead, the event loop didn't return, and the application would run termination handlers and exit itself.
+  //   [NSApp terminate:self];
```

Comment:
Please don't commit commented out code - we have Git to store the history.

Reply:
I'm aware of that but it's naive to think git will help someone in the immediate future if it doesn't do what they expect. I have removed it.